src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,175 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkingCapacity(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> oldNode = getNode(index);
+        T oldValue = oldNode.element;
+        oldNode.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return unlink(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> x = head; x != null; x = x.next) {
+            if (x.element == object) {
+                unlink(x);
+                return true;
+            }
+            if (object != null && object.equals(x.element)) {
+                unlink(x);
+                return true;
+            }
+        }
```

Comment:
Petro meant this but you should have asked him that, if you think you may have misunderstood something
```suggestion
            if (x.element == object || object != null && object.equals(x.element)) {
                unlink(x);
                return true;
            }
        }
```